### PR TITLE
setup deploy server

### PR DESCRIPTION
**Pull request by thapasya-m**
## PR Checklist
Please check that your PR fulfills the following requirements:

- [x] The commit message follows [our guidelines](https://github.com/pesto-students/batch-10-tripen/wiki/Commit-message-guidelines)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] GIFs/Videos/Images added (compulsory for frontend changes)

## PR Type
What kind of change does this PR introduce? 

- [ ] Bugfix
- [ ] Feature
- [ ] Tests changed / added / updated
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] CI related changes
- [ ] Documentation content changes
- [ ] Infrastructure changes
- [ ] Dependency upgrade
- [ ] Other... Please describe:

## Issue Number: 
#

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

## Other information

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -51,7 +51,18 @@ jobs:
           key: v1-deps-server-cache-{{ checksum "./server/yarn.lock" }}
       - run:
           name: Test server
-          command: cd server && yarn && yarn test
+          command: cd server && yarn && yarn
```
Comment (p-racoon) [inline .circleci/config.yml]: I wonder if there is a way we can avoid using yarn or npm?
Nothing to change, but let's explore this on a later date.

File: server/Procfile [hunk 2]
```diff
@@ -0,0 +1 @@
+web: yarn start
```
Comment (p-racoon) [inline server/Procfile]: would definitely need KT for this.
Just wanted to get your attention on filename `Procfile`, should it have been `.procfile` or something?

File: server/src/configs/env.js [hunk 3]
```diff
@@ -7,4 +7,4 @@ const configs = {
   clientUrl: process.env.CLIENT_URL || 'http://localhost:3000',
```
Comment (p-racoon) [inline server/src/configs/env.js]: wouldn't it be better if we directly connect the 'tripen-dev.netlify.com' here?

File: server/Procfile (hunk 2 above)
Comment (thapasya-m) [reply server/Procfile]: [Procfile](https://devcenter.heroku.com/articles/procfile) is correct.

File: server/src/configs/env.js (hunk 3 above)
Comment (thapasya-m) [reply server/src/configs/env.js]: we will store that URL in .env file. instead of assigning the url itself here.

File: .circleci/config.yml [hunk 4]
```diff
@@ -52,6 +52,18 @@ jobs:
...
+    docker:
+      - image: circleci/node:10.16.3
+    steps:
+      - checkout
+      - restore_cache:
+          key: v1-deps-server-cache-{{ checksum "./server/yarn.lock" }}
+      - deploy:
+          name: Deploy to heroku server
+          command:
+            git subtree push --prefix server https://heroku:$HEROKU_API_KEY@git.heroku.com/$HEROKU_APP_NAME.git master
```
Comment (nimish-gupta) [inline .circleci/config.yml]: Have you guys created only a single environment? Is anything on master is in production env? Is there any staging environment?

Comment (thapasya-m) [reply .circleci/config.yml]: Have you guys created only a single environment? 
yes.
Is anything on master is in production env?
yes.
Is there any staging environment?
no.

**Pull request merged**